packages/x-date-pickers/src/PickersTextField/PickersInputBase/PickersInputBase.tsx

```diff
@@ -292,6 +319,42 @@ const PickersInputBase = React.forwardRef(function PickersInputBase(
 
   const InputSectionsContainer = slots?.input || PickersInputBaseSectionsContainer;
 
+  const elementsIdList = React.useMemo(
+    () => elements.map((element) => element.content.id).join(','),
+    [elements],
+  );
+  const isSingleInputRange = React.useMemo(
+    () => elements.some((element) => element.content['data-range-position'] !== undefined),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [elementsIdList],
+  );
+  React.useEffect(() => {
+    if (isSingleInputRange) {
```

Comment:
```suggestion
    if (isSingleInputRange && ownerState.isPickerOpen) {
```

We don't show the highlight when the picker is not open 🤔 Might as well skip maybe?

Reply:
Great spot and a superb suggestion. 👍